src/pki/est/est_tls.c

```diff
@@ -321,7 +327,22 @@ static bool est_tls_init(private_est_tls_t *this, char *uri,
 		*path_str = '\0';
 	}
 
-	/* duplicate <hostname:port> string since we are going to manipulate it */
+	/* duplicate <label> string since we are going to manipulate it */
+	label_str = strdup(label);
+	size_t label_len = strlen(label_str);
+
+	/* ensure sure label ends with '/' character */
+	if (label_str[label_len - 1] == '/') {
+		this->http_label = strdup(label_str);
+	} else {
+		char suffix[] = {"/"};
+		char *res = malloc(label_len + 1);
+		strcpy(res, label_str);
+		strcat(res, suffix);
+		this->http_label = strdup(res);
+	}
+
```

Comment:
Thank you for the feedback and the details on possible error scenarios.

It is impressive how many issues one can introduced within a few lines of code in a language without exprience.

I'll try to fix all of that within the next week.

Reply:
Yeah, C is quite tricky in that regard.

> I'll try to fix all of that within the next week.

Great. Note that besides ensuring that the label ends with a `/` it might also be a good idea to strip any `/` at the start of it to avoid URLs with double slashes (i.e. something like `.../.well-known/est//<label>/<op>`).

Reply:
Hi @tobiasbrunner,

I've adjusted the label part as suggested:
- removed unnecessary `strdup()`
- used `asprintf` (which makes it ways more straight forward)
- checking the first character to avoid double `//`

The PR was rebased on the current `master`.

Thank you for your helpful and instructive feedback. :+1: